src/bin/hms-mqtt-publish/main.rs

```diff
@@ -83,7 +84,30 @@ fn main() {
         output_channels.push(Box::new(SimpleMqtt::<RumqttcWrapper>::new(&config)));
     }
 
+    if config.smiles_cooperation {
```

Comment:
A suggestion here would be to make the parameter optional. This way the option is introduced in a backwards compatible way, i.e. the parameter becomes an `Option<bool>` and one could continue using an existing config file with the new code.

```suggestion
    if config
        .smiles_cooperation
        .is_some_and(|value| value) {
            info!("S-Miles cloud cooperative mode enabled");
        }
    {
```

The loop below then would use a `if` statement.